core/src/test/scala/org/apache/spark/scheduler/TaskSetManagerSuite.scala

```diff
@@ -1778,6 +1778,48 @@ class TaskSetManagerSuite extends SparkFunSuite with LocalSparkContext with Logg
     assert(manager.resourceOffer("exec1", "host1", ANY).isEmpty)
   }
 
+  private def testSingleTaskSpeculation(singleTaskEnabled: Boolean): Unit = {
+    sc = new SparkContext("local", "test")
+    // Set the speculation multiplier to be 0 so speculative tasks are launched immediately
+    sc.conf.set(config.SPECULATION_MULTIPLIER, 0.0)
+    sc.conf.set(config.SPECULATION_ENABLED, true)
+    sc.conf.set(config.SPECULATION_SINGLETASKSTAGE_ENABLED, singleTaskEnabled)
+    // Set the threshold to be 60 minutes
+    sc.conf.set(config.SPECULATION_SINGLETASKSTAGE_DURATION_THRESHOLD.key, "60min")
```

Comment:
nit: why not keep it the same with the default value?

Reply:
This would validate that setting the conf actually works. IMO it's better than testing the default value. I can add another test to test the default value is 30 if that's preferred.

Reply:
sounds good